### PR TITLE
📝 Towers with ExternalTasks

### PR DESCRIPTION
**Pull request by LeoTT**
This task adds an addition to the previous tower example.

In the previous example script tasks were used to represent and manipulate the towers.
This PR simplifies the diagrams with external tasks.
The example is meant to underline the extensibility of BPMN-diagrams.

---

**Changes:**

1. Adds external task worker(nodejs)
1. Adds documentation for diagram changes and programming part.

**Issues:**

None

PR: #31

---

See the [examples](#example) for inspiration.

## How can others test the changes?

> Describe how others can test your changes (you can remove this section for typo fixes etc.)

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

## Example

<details>
<summary>
:warning: Before merging please click here to expand and follow the guide.
</summary>
<br>

Please use `:twisted_rightwards_arrows:` at the beginning of your merge commit title.

Example 1:

<pre>
<code>
:twisted_rightwards_arrows: :bug: Fix Wrong Text Decoration at ...
</code>
</pre>

To get your commit message, just copy the first part of this pull request.

Example 2:
<pre>
<code>
**Changes:**

- Fixes Wrong Text Decoration at ...
- Fixes some typos
- ...

**Issues:**

Closes #NumberOfFixedIssue

PR: #NumberOfThisPR
</code>
</pre>
</details>

## Emojis

<details>
<summary>
Expand for a list of most used Emojis.
</summary>
<br>

Please prefix your commit messages with an Emoji.

Ref: https://gitmoji.carloscuesta.me/

| Description              | Glyphe               | Emoji  |
|--------------------------|----------------------|--------|
| Bugfix                   | `:bug:`              | 🐛     |
| Configuration releated   | `:wrench:`           | 🔧     |
| Cosmetic                 | `:lipstick:`         | 💄     |
| Dependencies Downgrade   | `:arrow_down:`       | ⬇️     |
| Dependencies Upgrade     | `:arrow_up:`         | ⬆️     |
| Formatting               | `:art:`              | 🎨     |
| Improving Performance    | `:zap:`              | ⚡️     |
| Initial commit           | `:tada:`             | 🎉     |
| Linter                   | `:rotating_light:`   | 🚨     |
| Miscellaneous            | `:package:`          | 📦     |
| New Feature              | `:sparkles:`         | ✨     |
| Refactoring Code         | `:recycle:`          | ♻️     |
| Releasing / Version tags | `:bookmark:`         | 🔖     |
| Removing Stuff           | `:fire:`             | 🔥     |
| Tests                    | `:white_check_mark:` | ✅     |
| Work In Progress (WIP)   | `:construction:`     | 🚧     |

</details>

### CODE REVIEW

File: solutions/TowersExternalTask/nodejs/tower-service.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+      const emptySlotsLength = maxHeight - tower.length;
+      const emptySlots = Array(emptySlotsLength).fill(' ');
+      return emptySlots.concat(tower);
+    });
+
+    let lines = '';
+    for (let i = 0; i < maxHeight; i++) {
+
+      const newLine = sameLengthTowers
+            .reduce((text, tower) => `${text} ${tower[i]}`, '');
```
Comment (ElRaptorus) [inline solutions/TowersExternalTask/nodejs/tower-service.js]: Imho etwas zu viel Einrückung:

Besser so:

```ts
const newLine = 
  sameLengthTowers.reduce((text, tower) => `${text} ${tower[i]}`, '');
```

oder so:

```ts
const newLine = sameLengthTowers
  .reduce((text, tower) => `${text} ${tower[i]}`, '');
```

File: solutions/TowersExternalTask/nodejs/tower-service.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      lines = `${lines}\n${newLine}`
+    }
+
+    const underscore =' ' + '-'.repeat(sameLengthTowers.length * 2 - 1);
+    const numbers = Array.from(sameLengthTowers.keys())
+          .reduce((acc, cur) => acc + ' ' + cur);
```
Comment (ElRaptorus) [inline solutions/TowersExternalTask/nodejs/tower-service.js]: Wie oben, besser so:

```ts
const numbers = Array
  .from(sameLengthTowers.keys())
  .reduce((acc, cur) => acc + ' ' + cur);
```

File: solutions/TowersExternalTask/nodejs/tower-service.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        + underscore + '\n '
+        + numbers + '\n'
+        + subtitle + '\n\n';
+
+    console.log(output)
+  }
+}
+
+
+module.exports.TowerService = TowerService;
```
Comment (ElRaptorus) [inline solutions/TowersExternalTask/nodejs/tower-service.js]: Vllt. wäre es keine schlechte Idee den Code nach TypeScript zu überführen? 

Dann kann man die `export class` notation benutzen, egal in welche .js Version transpiliert wird.

Comment (LeoTT) [reply solutions/TowersExternalTask/nodejs/tower-service.js]: Guter Einwand.
Hatte JS benutzt, weil jeder TS-Entwickler JS kennt, aber andersrum nicht.

Auf Dauer wollte ich alle Programmiersprachen zeigen, für die wir ExternalTask Bibliotheken anbieten.
Dann könnt man JS, TS und Python zur Auswahl bieten.

**Pull request merged**